crates/oxc_linter/src/rules/react/iframe_missing_sandbox.rs

```diff
@@ -0,0 +1,227 @@
+    /// <iframe/>;
+    /// <iframe sandbox="invalid-value" />;
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```jsx
+    /// <iframe sandbox="" />;
+    /// <iframe sandbox="allow-origin" />;
+    /// ```
+    IframeMissingSandbox,
+    correctness
+);
+
+impl Rule for IframeMissingSandbox {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        match node.kind() {
+            AstKind::JSXOpeningElement(jsx_el) => {
+                let JSXElementName::Identifier(identifier) = &jsx_el.name else {
+                    return;
+                };
+
+                let name = identifier.name.as_str();
+                if name != "iframe" {
+                    return;
+                }
+
+                has_jsx_prop_ignore_case(jsx_el, "sandbox").map_or_else(
+                    || {
+                        ctx.diagnostic(missing_sandbox_prop(identifier.span));
+                    },
+                    |sandbox_prop| {
+                        validate_sandbox_attribute(sandbox_prop, ctx);
+                    },
+                );
+            }
+            AstKind::CallExpression(call_expr) => {
+                if is_create_element_call(call_expr) {
+                    let Some(Argument::StringLiteral(str)) = call_expr.arguments.first() else {
+                        return;
+                    };
+
+                    if str.value.as_str() != "iframe" {
+                        return;
+                    }
+
+                    if let Some(Argument::ObjectExpression(obj_expr)) = call_expr.arguments.get(1) {
+                        obj_expr
+                            .properties
+                            .iter()
+                            .find_map(|prop| {
+                                if let ObjectPropertyKind::ObjectProperty(prop) = prop {
+                                    if prop.key.is_specific_static_name("sandbox") {
+                                        return Some(prop);
+                                    }
+                                }
+
+                                None
+                            })
+                            .map_or_else(
+                                || {
+                                    ctx.diagnostic(missing_sandbox_prop(obj_expr.span));
+                                },
+                                |sandbox_prop| {
+                                    validate_sandbox_property(sandbox_prop, ctx);
+                                },
+                            );
+                    } else {
+                        ctx.diagnostic(missing_sandbox_prop(call_expr.span));
+                    }
+                }
+            }
+            _ => {}
+        }
+    }
+}
+fn validate_sandbox_value(literal: &StringLiteral, ctx: &LintContext) {
+    let attrs = literal.value.as_str().split(' ').collect::<Vec<&str>>();
+    let mut allow_same_origin = false;
+    let mut allow_scripts = false;
```

Comment:
nit: i'd call these `has_allow_same_origin` and `has_allow_scripts` to be clear about what they're referring to